.github/workflows/validate.yml

```diff
@@ -0,0 +1,48 @@
+name: Validate
+
+on: [push, pull_request, workflow_dispatch]
+
+jobs:
+  lint:
+    name: Lint
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [lts/*]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install Dependencies
+        run: npm ci
+      - name: Run Linting
+        run: npm run lint
+      - name: Inspect Lockfile
+        run: npm run lint:lockfile
+  test:
+    name: Snapshot Test
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        node-version: [lts/*]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ matrix.node-version }}
+      - name: Install Dependencies
+        run: npm ci
+      - name: Build
+        run: npm run build
+      - name: Run Snapshot Tests
+        run: npm test
+      - name: Archive Unit Test Coverage
+        uses: actions/upload-artifact@v3
```

Comment:
this is just creating a zip it is not posting to codecov

Reply:
Aha. I can port in from what we do in the agent.

Reply:
sounds good. it looks like the legacy circle CI plan used a similar action like we do to upload the coverage

Reply:
Looks like we're getting uploads now.